site/src/pages/tutorials/tutorial-blog-rest-step6.mdx

```diff
@@ -0,0 +1,152 @@
+  public HttpResponse deleteBlogPost(@Param int id) {
+    // Implementation
+  }
+  ```
+
+## 3. Implement service code
+
+In this step, write the code to delete a blog post, handle an exception, and block the operation.
+
+- [Delete a blog post](#delete-a-blog-post)
+- [Handle exceptions](#handle-exceptions)
+- [Add blocking](#add-blocking)
+
+### Delete a blog post
+
+Deleting a given blog post in this tutorial means removing a blog post from the map, `blogPosts`. However, in real services you would be performing this action on a database.
+
+To delete a blog post, copy the code below to the `deleteBlogPost()` method.
+
+```java filename=BlogService.Java highlight=2
+public HttpResponse deleteBlogPost(@Param int id) {
+  BlogPost removed = blogPosts.remove(id);
+  ...
+```
+
+### Handle exceptions
+
+What if there is no such post to delete? We can check if the blog exists before attempting to remove the blog post, but here, let's handle it after the attempt.
+
+1. Throw an `IllegalArgumentException` when no blog post exists with a given ID.
+  ```java filename=BlogService.java
+  if (removed == null) {
+    throw new IllegalArgumentException("The blog post does not exist. id: " + id);
+  }
+  ```
+2. Create an exception handler for the blog service:
+  1. Create a file, BadRequestExceptionHandler.java
+  2. In the file, declare a custom exception handler implementing Armeria's <type://ExceptionHandlerFunction> interface.
+  ```java filename=BadRequestExceptionHandler.java
+  public class BadRequestExceptionHandler implements ExceptionHandlerFunction {
+    private static final ObjectMapper mapper = new ObjectMapper();
+    ...
+  }
+  ```
+3. Implement your own exception handler by overriding the default `handleException()` method. Add a code block for handling the `IllegalArgumentException` thrown. For this tutorial, return a BAD REQUEST as the response.
+  ```java filename=BadRequesExceptionHandler.java highlight=3,9,10,12,14
+  @Override
+  public HttpResponse handleException(ServiceRequestContext ctx, HttpRequest req, Throwable cause) {
+    if (cause instanceof IllegalArgumentException) {
+        String message = cause.getMessage();
+        ObjectNode objectNode = mapper.createObjectNode();
+        objectNode.put("error", message);
+        try {
+            return HttpResponse.of(HttpStatus.BAD_REQUEST,
+                                    MediaType.JSON_UTF_8, mapper.writeValueAsString(objectNode));
+        } catch (JsonProcessingException e) {
+            return HttpResponse.of(HttpStatus.BAD_REQUEST);
+        }
+    }
+    return ExceptionHandlerFunction.fallthrough();
+  }
+  ```
+4. Assign the exception handler to the `deleteBlogPost()` method by annotating the `deletePost()` method with the <type://@ExceptionHandler> as follows.
+  ```java filename=BlogService.java highlight=1
+  @Delete("/blogs/:id")
+  @ExceptionHandler(BadRequestExceptionHandler.class)
+  public HttpResponse deleteBlogPost(@Param int id) {
+    ...
```

Comment:
```suggestion
  public HttpResponse deleteBlogPost(@Param int id) {
    ...
  }
```

Reply:
or
```suggestion
  public HttpResponse deleteBlogPost(@Param int id) { ... }
```

Reply:
Opted the second suggestion, thank you.
34b2de0ac1c4ddf269c992308bf94c30353de7ec